tests/models/dac/test_modeling_dac.py

```diff
@@ -0,0 +1,563 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Testing suite for the PyTorch Dac model."""
+
+import inspect
+import os
+import tempfile
+import unittest
+from typing import Dict, List, Tuple
+
+import numpy as np
+from datasets import Audio, load_dataset
+
+from transformers import AutoProcessor, DacConfig, DacModel
+from transformers.testing_utils import is_torch_available, require_torch, slow, torch_device
+
+from ...test_configuration_common import ConfigTester
+from ...test_modeling_common import ModelTesterMixin, _config_zero_init, floats_tensor
+from ...test_pipeline_mixin import PipelineTesterMixin
+
+
+if is_torch_available():
+    import torch
+
+
+def prepare_inputs_dict(
+    input_ids=None,
+    input_values=None,
+    decoder_input_ids=None,
+):
+    if input_ids is not None:
+        encoder_dict = {"input_ids": input_ids}
+    else:
+        encoder_dict = {"input_values": input_values}
+
+    decoder_dict = {"decoder_input_ids": decoder_input_ids} if decoder_input_ids is not None else {}
+
+    return {**encoder_dict, **decoder_dict}
+
+
+@require_torch
+class DacModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=5,
+        num_channels=1,
+        is_training=False,
+        intermediate_size=15992,
```

Comment:
The hidden-dims used here are way too big for the fast tests. We should use the smallest possible version of the model that lets us run these tests. This ensures the fast tests actually run **fast**! Remember that there are over 200 models in the Transformers library -> we need each model tester to run in << 30-seconds to keep the speed of the CI reasonable. This is primarily achieved by using super small models and the lowest possible batch sizes.

See the dims used in EnCodec as an example of the order of magnitude these dims should be: https://github.com/huggingface/transformers/blob/5e89b335ab088b52e5dbb50c5368d944e9052ed4/tests/models/encodec/test_modeling_encodec.py#L76

Reply:
Oh ok I see, that makes sense :)